### PR TITLE
Update of Git

### PR DESCRIPTION
**Pull request by weifanfrank**
I have not finished yet, just saving this page first for backup.

### CODE REVIEW

**Comment by weifanfrank** [comment]
.

**Comment by mjwen** [comment]
The markdown format looks great and the context is almost there. Here are a few comments:

1. Think of you are teaching somebody who has never used it, and what you want is to get the person familiar with the entire process of basic git. A good approach is what we discussed the other day: walking through the entire process of turning a non-git directory to a git directory. So, you might want to start with `git init`. 
2. It is not helpful to simply list the commands, and what they do. It's more useful to link them together in a logic way: from `git init` to `git push`. This is related to comment 1. So, probably start with creating an empty directory, then adding a file into it, initializing the file, committing it ...  
3. Commands like `git add` should be `git add <filename>`, and you need to example that `<filename>` the the file you wan to commit. Additionally you can say if you want add all files in the current directory, you can do `git  add .`. Every commands need to be explained a bit (using very brief language).

File: docs/devtool/git.md [hunk 1]
```diff
@@ -1,3 +1,97 @@
...
+
+## Set up the Git environment
+Once Git is installed, you'll need to set up your environment by configuring your username and email address, so that Git can attribute your changes to you. To do this, open a command prompt or terminal window and type the following commands, replacing "Your Name" and "Youremail" with your own information:
+```
+$ git config --global user.name "Your Name"
+$ git config --global user.email "Youremail@example.com"
+```
+
+## Git workflow and basic commands
+<img width="820" alt="image" src="https://user-images.githubusercontent.com/123142931/219302278-e9655f60-d6b6-4594-a0fc-4a649c6e11d0.png">
```
Comment (mjwen) [inline docs/devtool/git.md]: Can you please place the image in the `group_manual/docs/image/` directory, and using the figure syntax as in [this file](https://raw.githubusercontent.com/wengroup/group_manual/main/docs/contributing/edit.md) to refer to it?

File: docs/devtool/git.md [hunk 2]
```diff
@@ -1,3 +1,97 @@
...
+
+
+1. Create a new repository to start using Git.
+```
+$ git init
+```
+
+2-1. Add files from workspace(working directory) to staging area(first level).
```
Comment (mjwen) [inline docs/devtool/git.md]: This is not a perfect syntax for markdown -- we typically do not use stuff like 2-1. I'd suggest replacing them subtitles and bulletin points. More specific,  `1.`, `2.`.. can be conveyed to `###` subtitles, and `2-1`, `2-2`, can be converted to bulletin using `-`.

File: docs/devtool/git.md [hunk 3]
```diff
@@ -1,3 +1,97 @@
...
+```
+$ git add <filename>
+```
+
+2-2. You can also use 'git add .' to add all files in the current directory to the staging area.
+```
+$ git add .
+```
+
+3-1. Transfer files from staging(first level) to repository(second level).
```
Comment (mjwen) [inline docs/devtool/git.md]: You need to add a space between `staging` and `(`. Same for all other opening parenthesis in the file.

File: docs/devtool/git.md [hunk 4]
```diff
@@ -1,3 +1,97 @@
...
+```
+$ git push <filename>
+
+```
+5. Transfer files from remote repository(Github) to the workspace.
+```
+$ git pull <filename>
+```
+
+6. Allow you to retrieve changes from a remote repository without merging them into your local repository(branch). This means that you can review the changes and decide whether to merge them into your branch or not. This gives you more control over the merging process and can help you avoid conflicts when multiple contributors are working on the same codebase.
```
Comment (mjwen) [inline docs/devtool/git.md]: Remove `Allow you to`, and start directly with `Retrieve`. Same for point 7.

Comment (weifanfrank) [reply docs/devtool/git.md]: Done

File: docs/devtool/git.md (hunk 3 above)
Comment (weifanfrank) [reply docs/devtool/git.md]: Done

File: docs/devtool/git.md (hunk 2 above)
Comment (weifanfrank) [reply docs/devtool/git.md]: Done

File: docs/devtool/git.md (hunk 1 above)
Comment (weifanfrank) [reply docs/devtool/git.md]: Done

**Comment by mjwen** [comment]
@weifanfrank all looks good. Thanks!

**Pull request merged**